Prototope/Behaviors.swift

```diff
@@ -0,0 +1,194 @@
+    
+    init(hostLayer: Layer, config: CollisionBehavior) {
+        self.config = config
+        self.previousState = CollisionState.stateForLayer(hostLayer, andLayer: self.config.otherLayer)
+        
+        super.init(hostLayer: hostLayer)
+    }
+
+    override func update() {
+        self.updateWithState(CollisionState.stateForLayer(self.hostLayer, andLayer: self.config.otherLayer))
+    }
+    
+    func updateWithState(state: CollisionState) {
+        let old = previousState
+        switch (self.config.kind, old, state) {
+            
+        case (.Entering,.NonOverlapping,.PartiallyIntersects):
+            fallthrough
+        case (.Entering,.NonOverlapping,.ContainedIn):
+            fallthrough
+        case (.Entering,.NonOverlapping,.Contains):
+            fire()
+            
+        case (.Leaving,.PartiallyIntersects,.NonOverlapping):
+            fallthrough
+        case (.Leaving,.ContainedIn,.NonOverlapping):
+            fallthrough
+        case (.Leaving,.Contains,.NonOverlapping):
+            fire()
+            
+        default:
+            ()
+        }
+        
+        self.previousState = state
+    }
+    
+    func fire() {
+        self.config.handler()
+    }
+}
+
+//MARK: Global Behavior Store
+
+// Global bag of layer behavior state
+class BehaviorStore {
+    var heartbeat: Heartbeat!
+    
+    var registeredBindings: Set<BehaviorBinding> {
+        didSet {
+            self.heartbeat.paused = self.registeredBindings.count == 0
+        }
+    }
+    
+    init() {
+        self.registeredBindings = Set<BehaviorBinding>()
+        
+        self.heartbeat = Heartbeat { [unowned self] _ in
+            self.tick()
+        }
+    }
+    
+    deinit {
+        self.heartbeat.stop()
+    }
+    
+    func tick() {
+        for b in self.registeredBindings {
+            b.update()
+        }
+    }
+    
+    func updateWithLayer(layer: Layer, behaviors: [BehaviorType]) {
+        var knownBindings = lazy(self.registeredBindings).filter { $0.hostLayer == layer }
+        
+        let otherBindings = self.registeredBindings.subtract(knownBindings)
+        
+        let newBindings = behaviors.map { b -> BehaviorBinding in
+            return self.createBindingForLayer(layer, behavior: b)!
+        }
+        
+        self.registeredBindings = otherBindings.union(newBindings)
+    }
+    
+    func registerBinding(binding: BehaviorBinding) {
+        self.registeredBindings.insert(binding)
+    }
+    
+    func unregisterBinding(binding: BehaviorBinding) {
+        self.registeredBindings.remove(binding)
+    }
+    
+    func createBindingForLayer(layer: Layer, behavior: BehaviorType) -> BehaviorBinding? {
+        if let collisionBehavior = behavior as? CollisionBehavior {
+            return CollisionBehaviorBinding(hostLayer: layer, config: collisionBehavior)
+        }
+        
+        return nil
+    }
+}
+
+var globalBehaviorStore = BehaviorStore()
```

Comment:
> With new live-reload features, we've gotta be more careful about globals. I know we've got a few more to clean up, but let's at least not add more. :)

Yup, this was implemented before I had a look at the shiny new stuff. 

I’ll do another pass on this PR to bring it up to date and incorporate any feedback :+1: